device/tt_core_coordinates.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2024 Tenstorrent Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+ #include "device/tt_xy_pair.h"
+
+
+/*
+ * CoordSystem is an enum class that represents all types of coordinate
+ * systems that can be used to represent a core's location.
+ * This is used both for V1 and V2.
+ */
+enum class CoordSystem {
+    LOGICAL,
+    PHYSICAL,
+    VIRTUAL,
+    TRANSLATED,
+};
+
+// ************************************************************************************************
+// V1
+
+/*
+ * CoreType is an enum class that represents all types of cores
+ * present on the Tenstorrent chip.
+ */
+enum class CoreType {
+  ARC,
+  DRAM,
+  ETH,
```

Comment:
I think we want to expose the concept of both active and idle eth from the bottom of the stack

Reply:
yes, active is "has links" and inactive is "no links".  active is running the "base firmware" to handle host read/write tunneling, "inactive" does not.  we run workloads on both types, but different types of workloads.

Reply:
Follow up issue for getting active ethernet cores https://github.com/tenstorrent/tt-umd/issues/344